tests/unit/test-data.js

```diff
@@ -0,0 +1,324 @@
+  MAXIMUM_BET: '200',
+  MINIMUM_BET: '100',
+  BET_POOL: '1000',
+},
+{
+  CARD_NUMBER: '8',
+  SUIT: 'diamonds',
+  GAME_ID: '2',
+  ROUND: 'river',
+  MAXIMUM_BET: '2000',
+  MINIMUM_BET: '1000',
+  BET_POOL: '104000',
+},
+{
+  CARD_NUMBER: '9',
+  SUIT: 'diamonds',
+  GAME_ID: '2',
+  ROUND: 'river',
+  MAXIMUM_BET: '2000',
+  MINIMUM_BET: '1000',
+  BET_POOL: '104000',
+},
+{
+  CARD_NUMBER: '10',
+  SUIT: 'diamonds',
+  GAME_ID: '2',
+  ROUND: 'river',
+  MAXIMUM_BET: '2000',
+  MINIMUM_BET: '1000',
+  BET_POOL: '104000',
+},
+{
+  CARD_NUMBER: 'J',
+  SUIT: 'diamonds',
+  GAME_ID: '2',
+  ROUND: 'river',
+  MAXIMUM_BET: '2000',
+  MINIMUM_BET: '1000',
+  BET_POOL: '104000',
+},
+{
+  CARD_NUMBER: 'Q',
+  SUIT: 'diamonds',
+  GAME_ID: '2',
+  ROUND: 'river',
+  MAXIMUM_BET: '2000',
+  MINIMUM_BET: '1000',
+  BET_POOL: '104000',
+},
+{
+  CARD_NUMBER: null,
+  SUIT: null,
+  GAME_ID: '3',
+  ROUND: 'river',
+  MAXIMUM_BET: '20000',
+  MINIMUM_BET: '10000',
+  BET_POOL: '1040000',
+}];
+
+const mergedRawGames = [{
+  CARD_NUMBER: '2',
+  SUIT: 'spades',
+  GAME_ID: '1',
+  ROUND: 'blind',
+  MAXIMUM_BET: '200',
+  MINIMUM_BET: '100',
+  BET_POOL: '1000',
+  tableCards: [{ cardNumber: '2', cardSuit: 'spades' }],
+},
+{
+  CARD_NUMBER: '8',
+  SUIT: 'diamonds',
+  GAME_ID: '2',
+  ROUND: 'river',
+  MAXIMUM_BET: '2000',
+  MINIMUM_BET: '1000',
+  BET_POOL: '104000',
+  tableCards:
+ [{ cardNumber: '8', cardSuit: 'diamonds' },
+   { cardNumber: '9', cardSuit: 'diamonds' },
+   { cardNumber: '10', cardSuit: 'diamonds' },
+   { cardNumber: 'J', cardSuit: 'diamonds' },
+   { cardNumber: 'Q', cardSuit: 'diamonds' }],
+},
+{
+  CARD_NUMBER: null,
+  SUIT: null,
+  GAME_ID: '3',
+  ROUND: 'river',
+  MAXIMUM_BET: '20000',
+  MINIMUM_BET: '10000',
+  BET_POOL: '1040000',
+  tableCards: [],
+}];
+
+const serializedGames = {
+  links: { self: '/v1/games' },
+  data:
+ [{
+   type: 'game',
+   id: '1',
+   links: { self: '/v1/games/1' },
+   attributes:
+      {
+        round: 'blind',
+        minimumBet: 100,
+        maximumBet: 200,
+        betPool: 1000,
+        tableCards: [{ cardNumber: '2', cardSuit: 'spades' }],
+      },
+ },
+ {
+   type: 'game',
+   id: '2',
+   links: { self: '/v1/games/2' },
+   attributes:
+      {
+        round: 'river',
+        minimumBet: 1000,
+        maximumBet: 2000,
+        betPool: 104000,
+        tableCards:
+         [{ cardNumber: '8', cardSuit: 'diamonds' },
+           { cardNumber: '9', cardSuit: 'diamonds' },
+           { cardNumber: '10', cardSuit: 'diamonds' },
+           { cardNumber: 'J', cardSuit: 'diamonds' },
+           { cardNumber: 'Q', cardSuit: 'diamonds' }],
+      },
+ },
+ {
+   type: 'game',
+   id: '3',
+   links: { self: '/v1/games/3' },
+   attributes:
+      {
+        round: 'river',
+        minimumBet: 10000,
+        maximumBet: 20000,
+        betPool: 1040000,
+        tableCards: [],
+      },
+ }],
+};
+const serializedGame = {
+  links: { self: '/v1/games/1' },
```

Comment:
`/v1/games` was reused many times. You should store it somewhere and reuse it.

Reply:
It's just the test data to be used to compared. I don't see reasons to refer to a variable in the test data.

Reply:
Test data is also part of your code, it’s mainly for the good coding practice.

Reply:
To me it's just aliasing instead of refactoring the code. It's just a dummy data and changing it would not affect the test at all.

Reply:
It's totally your choice to decide how to write your code since no one else will need to maintain this API besides yourself.

My question is what's the reason not to do that even it's just an alias to you? How can you guarantee we won't need to change this string in the future? If that's the case, do we wanna find them all and replace them one by one? Since we're working as a group, a thing we need to always keep in mind is your code might need to be read and maintained by others in the future. If your code can be clearer and easier to maintain, why not doing it?

Reply:
Sounds good but wouldn't that rule extends to all other strings in the test data? What's the boundary between raw data and reusable data?

Reply:
I guess I'm not really sure what did you mean. If you were saying something like `diamonds`, 'spades', etc. those are constant and reused many times, then yes I'd agree to store them into constant strings to reuse. When we mentioned something like "this code was used many times", I think it's worthy to think about it more will there be a method to improve/combine/reuse the logic/function/variable to enhance your code. No matter it's in your test data or in the real code.
@ianbrown9475 since you're really good at writing clean code, can you also share some of your thought?

Reply:
I think @tsoliangwu0130 makes a really good point that the test data, even though it's just dummy data, is still code that we're committing to source control and being used by the unit tests. Even though it may seem like random data created for a simple purpose, we need to treat it like actual code, since it's still going to be maintained. For our other APIs, this means that other developers will have to eventually go back, look at the code, and modify it when there's a code/design change.

I would suggest writing these so that another developer can easily open this file for the first time and make changes as easily as possible. Variables that have values like `v1` or `/v1/games` should probably go at the very top of this file, with some documentation on them. This way, if the api version is bumped to `v2` for example, developers can open this file, see where this `v1` variable is used, and replace it once with `v2`. Then, they don't have to find all instances of `v1` and replace it manually.

For values like `diamonds`, `spades`, `A`, `2`, `Q`, `blind`, ...; these could be stored in a variable as well. Ideally, you would load these enum values from your openapi document, but that may not be necessary for this learning project. @nosnatef as far as when to refactor these values, I think it's important to think about how easy the test data would be to change if there was a design change.